hydroflow/examples/kvs_bench/server.rs

```diff
@@ -0,0 +1,272 @@
+                            },
+                            KvsRequest::_Get {key} => gets.give((key, addr)),
+                        }
+                    });
+
+                // broadcast out locally generated changes to other nodes.
+                client_input[broadcast]
+                    -> batch(batch_interval_ticker)
+                    -> group_by::<'tick>(|| Option::<MyLastWriteWinsRepr>::None, |accum: &mut Option<MyLastWriteWinsRepr>, reg: MyLastWriteWinsRepr| {
+                        // TODO: Could turn this into the BottomRepr lattice rather than having the external Option wrapping.
+                        // If multiple writes have hit the same key then they can be merged before sending.
+                        if accum.is_none() {
+                            *accum = Some(reg);
+                        } else {
+                            <MyLastWriteWins as Merge<MyLastWriteWins>>::merge(accum.as_mut().unwrap(), reg);
+                        }
+                    })
+                    -> filter_map(|(key, opt_reg)| opt_reg.map(|reg| (key, reg))) // to filter out bottom types since they carry no useful info.
+                    // -> inspect(|x| println!("{gossip_addr}:{:5}: sending to peers: {x:?}", context.current_tick()))
+                    -> for_each(|x| { transducer_to_peers_tx.send(x).unwrap(); });
+
+                // join for lookups
+                lookup = lattice_join::<'static, 'tick, MyLastWriteWins, SetUnion>();
+
+                client_input[store]
+                    // -> inspect(|x| println!("{gossip_addr}:{:5}: stores-into-lookup: {x:?}", context.current_tick()))
+                    -> [0]lookup;
+
+                // Feed gets into the join to make them do the actual matching.
+                client_input[gets]
+                    -> map(|(key, addr)| {
+                        let mut gset = HashSet::new();
+
+                        let seq_num = {
+                            let mut seq_borrow = get_seq_num.borrow_mut();
+                            *seq_borrow += 1;
+                            *seq_borrow
+                        };
+
+                        gset.insert((addr, seq_num));
+                        (key, gset)
+                    })
+                    // -> inspect(|x| println!("{gossip_addr}:{:5}: gets-into-lookup: {x:?}", context.current_tick()))
+                    -> [1]lookup;
+
+                // Send get results back to user
+                lookup
+                    -> map(|(key, (reg, gets))| {
+                        gets.into_iter().map(move |(dest, _seq_num)| {
+                            (KvsResponse::GetResponse { key, reg: reg.clone() }, dest)
+                        })
+                    })
+                    -> flatten()
+                    // -> inspect(|x| println!("{gossip_addr}:{:5}: Response to client: {x:?}", context.current_tick()))
+                    -> dest_sink(transducer_to_client_tx);
```

Comment:
!! This doesn't do the bounding/backpressure you want because `dest_sink` puts things into an unbounded channel before pushing into the `Sink`. Could be considered a `dest_sink` bug depending on what behavior we think is right

Reply:
hmm, I'm not sure I understand completely the issue here. I also forget the reason why I am using an unbounded channel to begin with, I think it was likely because there was some util function I wanted to use that was only defined for unbounded channels but I will try to do it with a bounded channel again to see what the original issue was.

Reply:
Other way around, you're using a bounded channel here which unfortunately doesn't really work the way youd expect

Reply:
Oh, I think I remember why it is using bounded_channel now, it's so you can use PollSender from tokio, so that you can then feed it into dest_sink. I must've written this part a long time ago, instead this can be changed to for_each(|x| channel.send(x)). Then it can be a bounded channel.

I'm still not sure I understand the backpressure issue though.